### ISSUE TITLE
Python does not build without WITH_THREADS defined on Windows/Visual Studio

### ISSUE THREAD

**Issue by 271e878f-1235-4b2e-b652-7a77057c8224** [python/cpython#74439]
BPO | [30253](https://bugs.python.org/issue30253)
--- | :---
Nosy | @pfmoore, @tjguk, @zware, @zooba

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2017-05-03.10:10:00.892>
labels = ['build', 'OS-windows']
title = 'Python does not build without WITH_THREADS defined on Windows/Visual Studio'
updated_at = <Date 2017-05-05.22:18:16.849>
user = 'https://bugs.python.org/steveire'
```

bugs.python.org fields:
```python
activity = <Date 2017-05-05.22:18:16.849>
actor = 'terry.reedy'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Build', 'Windows']
creation = <Date 2017-05-03.10:10:00.892>
creator = 'steveire'
dependencies = []
files = []
hgrepos = []
issue_num = 30253
keywords = []
message_count = 1.0
messages = ['292879']
nosy_count = 5.0
nosy_names = ['paul.moore', 'tim.golden', 'steveire', 'zach.ware', 'steve.dower']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue30253'
versions = ['Python 3.6']
```

</p></details>

**Comment by 271e878f-1235-4b2e-b652-7a77057c8224** [comment]
As there is no configuration system for python on Windows (bpo-30252) I tried to change pyconfig.h to comment out some lines:

 // #define NT_THREADS
 // #define WITH_THREAD

After building, I had to additionally patch 

* threadmodule.c and thread.c to exclude all content (I don't know if there is a way to exclude the file entirely with the Python build system - there should be something like that as part of a configure system).
* Wrap uses of PyThread_get_thread_ident() in ifdefs in timemodule.c. This is surprising because I would expect it to not compile on any platform without the ifdefs.
